doc/docusaurus/docs/delve-deeper/languages.md

```diff
@@ -81,7 +83,10 @@ Additionally, the desugaring process might transform code in such a way that it
 Furthermore, complications arise when the new language and the host language do not exactly agree on semantics or evaluation strategies.
 This disparity can lead to behaviors where the same code might act differently when compiled and executed in the host language versus the new language.
 It can also result in idioms that work well in the host language being inappropriate for the new language.
-For example, while guarded recursion is a useful idiom in Haskell, it is unsuitable for Plutus Tx due to Plutus Tx's use of call-by-value evaluation.
+For example, while guarded recursion is a useful idiom in Haskell, it might not be suitable for Plutus Tx due to Plutus Tx's use of call-by-value evaluation.
+
+Another drawback of using a subset of a language is that, determining whether a program conforms to the allowed subset typically doesn't happen at type checking time, but at target code generation time.
+This not only delays error detection cmpared to eDSLs, but makes it harder to produce clear error messages, since by target code generation time, the AST may have already been transformed and optimized, obscuring its connection to the original source code.
```

Comment:
```suggestion
This not only delays error detection compared to eDSLs, but makes it harder to produce clear error messages, since by target code generation time, the AST may have already been transformed and optimized, obscuring its connection to the original source code.
```